content/automate/ManageIQ/System/Event/EmsEvent/RHEVM.class/__methods__/update_vm_import_status.rb

```diff
@@ -0,0 +1,13 @@
+#
```

Comment:
@matobet, Here's a spec test I created for the event class method. It needs work, but it has the basics. Also, I had to modify the Automate method for the spec tests to run properly, it probably needs work/testing as well.  I hope this helps. 

reformatted method
`#
# Description: Update VM import status.
#

module ManageIQ
  module Automate
    module System
      module Event
        module EmsEvent
          module RHEVM
            class UpdateVmImportStatus
              def initialize(handle = $evm)
                @handle = handle
              end

              def main
                update_import_status(vm)
              end

              private

              def vm_id
                event_stream = @handle.root['event_stream']
                if event_stream.nil?
                  @handle.log(:error, 'event_stream not found')
                  raise 'event_stream not found'
                end
                event_stream.vm_or_template_id
              end

              def vm
                @handle.vmdb('Vm', vm_id).tap do |vm|
                  if vm.nil?
                    @handle.log(:error, 'VM object not found')
                    raise 'VM object not found'
                  end
                end
              end

              def import_status
                @handle.root['event_type'] == 'IMPORTEXPORT_IMPORT_VM_FAILED' ? 'failure' : 'success'
              end

              def update_import_status(vm)
                vm.custom_set(:import_status, import_status)
                exit MIQ_OK
              end
            end
          end
        end
      end
    end
  end
end

if __FILE__ == $PROGRAM_NAME
  ManageIQ::Automate::System::Event::EmsEvent::
    RHEVM::UpdateVmImportStatus.new.main
end
`
`require_domain_file

describe ManageIQ::Automate::System::Event::EmsEvent::
          RHEVM::UpdateVmImportStatus do

  let(:user)          { FactoryGirl.create(:user_with_group) }
  let(:miq_server)    { EvmSpecHelper.local_miq_server }
  let(:ems)           { FactoryGirl.create(:ems_vmware_with_authentication, :zone => FactoryGirl.create(:zone)) }
  let(:vm)            { FactoryGirl.create(:vm_vmware, :ext_management_system => ems) }
  let(:miq_event)     { FactoryGirl.create(:miq_event, :event_type => "some_event") }

  let(:ems_event) do
    FactoryGirl.create(:ems_event, :vm_or_template => vm, :ext_management_system => ems)
  end
  let(:svc_model_miq_server) { MiqAeMethodService::MiqAeServiceMiqServer.find(miq_server.id) }
  let(:svc_model_user) { MiqAeMethodService::MiqAeServiceUser.find(user.id) }

  let(:root_hash) do
    { 'event_stream' => MiqAeMethodService::MiqAeServiceEmsEvent.find(ems_event.id),
      'event_type'   => 'IMPORTEXPORT_IMPORT_VM_FAILED' }
  end

  let(:ae_service) do
    Spec::Support::MiqAeMockService.new(root_object).tap do |service|
      current_object = Spec::Support::MiqAeMockObject.new
      current_object.parent = root_object
      service.object = current_object
    end
  end

  let(:root_object) do
    obj = Spec::Support::MiqAeMockObject.new(root_hash)
    obj["user"] = svc_model_user
    obj["miq_server"] = svc_model_miq_server
    obj
  end

  it "method succeeds" do
    described_class.new(ae_service).main
  end
end
`